src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,244 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        https://urldefense.com/v3/__http://www.apache.org/licenses/LICENSE-2.0__;!!N30Cs7Jr!UgbMbEQ59BIK-1Xslc7QXYm0lQBh92qA3ElecRe1CF_9YhXxbwPOZa6j4plru7B7kCJ7bKQgHxgQrket3-Dnk268sIdA7Qb8$
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/scenes/ExtensionFieldsSets.h>
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/CommonIterator.h>
+
+namespace chip {
+namespace scenes {
+
+class SceneTable
+{
+public:
+    static constexpr size_t kIteratorsMax       = CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS;
+    static constexpr size_t kSceneNameMax       = ZCL_SCENES_CLUSTER_MAXIMUM_NAME_LENGTH;

```

Comment:
Why have `ZCL` macros. Since this is not currently a working feature, suggest moving away from this. This should be a CHIP_CONFIG_X